include/envoy/stream_info/filter_state.h

```diff
@@ -55,6 +55,69 @@ class FilterState {
             (dynamic_cast<const T*>(getDataGeneric(data_name)) != nullptr));
   }
 
+  /**
+   * The addToList, hasListWithName, forEachListItem are variants of the above
+   * functions, that operate on list data. Multiple elements could be added
+   * the list under the same data_name.
+   * @param data_name the name of the data being added to the end of a list.
+   * @param data an owning pointer to the data to be stored.
+   * Note that data_names for list elements have no relation to the data_names
```

Comment:
nit, suggestion: I read "have no relation" as "do not share the same namespace", which I think is a clearer phrasing, but I'm ok with this phrasing if you prefer it.

However (non-nit, non-suggestion): This strikes me as being potentially confusing to the user of this class.  I'd think it would be better if they shared a namespace and conflicts resulted in errors.  What do you think?

Reply:
I didn't seen any response to this comment?  Can you give a sense as to why you want the two stores in different namespaces?